test_scripts/AppIconsFolder/ATF_N_AppIconsAmountToRemove_larger_than_icons_amount.lua

```diff
@@ -0,0 +1,326 @@
+      returnValue = false
+    end
+    return returnValue
+end
+
+-- Get folder size
+local function dSize(PathToFolder)
+  local sizeFolderInBytes
+  local aHandle = assert( io.popen( "du -sh " ..  tostring(PathToFolder), 'r'))
+  local buff = aHandle:read( '*l' )
+  local sizeFolder, measurementUnits = buff:match("([^%a]+)(%a)")
+  if measurementUnits == "K" then
+    sizeFolder  =  string.gsub(sizeFolder, ",", ".")
+    sizeFolder = tonumber(sizeFolder)
+    sizeFolderInBytes = sizeFolder * 1024
+  elseif
+    measurementUnits == "M" then
+    sizeFolder  =  string.gsub(sizeFolder, ",", ".")
+    sizeFolder = tonumber(sizeFolder)
+    sizeFolderInBytes = sizeFolder * 1048576
+  end
+  return sizeFolderInBytes
+end
+
+-- Make AppIconsFolder full
+local function makeAppIconsFolderFull(AppIconsFolder)
+  local sizeToFull
+  local sizeAppIconsFolderInBytes = dSize(config.pathToSDL .. tostring(AppIconsFolder))
+  sizeToFull = 1048576 - sizeAppIconsFolderInBytes
+  local i =1
+  while sizeToFull > 326360 do
+    os.execute("sleep " .. tonumber(10))
+    local copyFileToAppIconsFolder = assert( os.execute( "cp files/icon.png " .. tostring(config.pathToSDL) .. tostring(AppIconsFolder) .. "/icon" .. tostring(i) ..".png"))
+    i = i + 1
+    if copyFileToAppIconsFolder ~= true then
+      commonFunctions:userPrint(31, " Files are not copied to " .. tostring(AppIconsFolder))
+    end
+    sizeAppIconsFolderInBytes = dSize(config.pathToSDL .. tostring(AppIconsFolder))
+    sizeToFull = 1048576 - sizeAppIconsFolderInBytes
+    if i > 50 then
+      commonFunctions:userPrint(31, " Loop is breaking due to a lot of iterations ")
+      break
+    end
+  end
+end
+
+local function checkFunction()
+  local applicationFileExistsResult = checkFileExistence(applicationFileToCheck)
+  local firstFileExistsResult = checkFileExistence(firstOldFileToCheck)
+  local secondFileExistsResult = checkFileExistence(secondOldFileToCheck)
+  local thirdFileExistsResult = checkFileExistence(thirdOldFileToCheck)
+  local aHandle = assert( io.popen( "ls " .. config.pathToSDL .. "Icons/" , 'r'))
+  local listOfFilesInStorageFolder = aHandle:read( '*a' )
+  commonFunctions:userPrint(32, "Content of storage folder: " ..tostring("\n" ..listOfFilesInStorageFolder))
+
+  if applicationFileExistsResult ~= true then
+    commonFunctions:userPrint(31, tostring(RAIParameters.appID) .. " icon is absent")
+    status = false
+  end
+  if 
+    firstFileExistsResult ~= false or
+    secondFileExistsResult ~= false or
+    thirdFileExistsResult ~= false then
+    commonFunctions:userPrint(31,"Oldest icons are not deleted from AppIconsFolder")
+    status = false
+  end
+    return status
+end
+
+--[[ Preconditions ]]
+commonSteps:DeleteLogsFileAndPolicyTable()
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test.Precondition_stopSDL()
+  StopSDL()
+end 
+
+function Test.Precondition_configureAppIconsFolder()
+  checkSDLPathValue()
```

Comment:
@aarotar 
As I can see you use function `checkSDLPathValue()` in alot of steps of this script. First of them add '/' to the and of `config.pathToSDL` value (in needed), next ones do nothing because `config.pathToSDL` already contains value with '/' at the end.
I propose remove all calls of function  `checkSDLPathValue()` from the script except first one.